### PR TITLE
Generate gh pages

### PR DESCRIPTION
**Pull request by gjasny**
@jupp0r  Could you please create a token named `GITHUB_TOKEN` as described [here](https://docs.travis-ci.com/user/deployment/pages/#setting-the-github-token) and add it to the global `secure` elements list in `.travis.yml`?

That should update doxygen on https://jupp0r.github.io/prometheus-cpp/
Example output available here: https://gjasny.github.io/prometheus-cpp/

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -48,6 +48,18 @@ script:
...
+  - doxygen
+  - touch html/.nojekyll
+
+deploy:
+  provider: pages
+  local-dir: doc/html
+  skip-cleanup: true
+  github-token: $GITHUB_TOKEN
+  on:
+    branch: master
```
Comment (jerryct) [inline .travis.yml]: I would not push to master and add generated files to the repository. The master branch is the one everybody checks out and works with. I would rather add another branch (like gh-pages) or simply create a second project under prometheus-cpp which only stores the generated files (like in my example).

File: .travis.yml [hunk 2]
```diff
@@ -48,6 +48,18 @@ script:
   - make DESTDIR=`pwd`/deploy install
   - popd
 
+  - cd "${TRAVIS_BUILD_DIR}/doc"
```
Comment (jerryct) [inline .travis.yml]: Better use `pushd` and call `popd` after line 53. That makes it extensible for the future without touching this 3 lines of code.

File: .travis.yml [hunk 3]
```diff
@@ -48,6 +48,20 @@ script:
   - make DESTDIR=`pwd`/deploy install
   - popd
 
+  - pushd .
+  - cd doc
```
Comment (jerryct) [inline .travis.yml]: only minor but you could write pushd ./doc and get rid of this extra cd

**Comment by jupp0r** [comment]
Sorry for responding so late, got back from vacation. I'll take care of this later this week. Thanks @gjasny

**Comment by jupp0r** [comment]
Should be unblocked by #250

**Comment by jupp0r** [comment]
Please rebase + merge

**Pull request merged**